hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/impl/CommitUtilsWithMR.java

```diff
@@ -49,10 +49,11 @@ private CommitUtilsWithMR() {
   /**
    * Get the location of magic job attempts.
    * @param out the base output directory.
+   * @param jobId unique Job ID.
    * @return the location of magic job attempts.
    */
-  public static Path getMagicJobAttemptsPath(Path out) {
-    return new Path(out, MAGIC);
+  public static Path getMagicJobAttemptsPath(Path out, String jobId) {
+    return new Path(out, MAGIC + "_" + jobId);
```

Comment:
Consider using "MAGIC PATH_PREFIX" here, receiving the actual job uuid as a parameter instead of the "job-uuid" formatted version.. this method formatJobDir(..) could be removed I think